plugins/become/doas.py

```diff
@@ -111,7 +111,14 @@ def build_become_command(self, cmd, shell):
 
         self.prompt = True
 
-        become_exe = self.get_option('become_exe') or self.name
+        become_exe = self.get_option('become_exe')
+        if not become_exe or become_exe == self.name:
+            # HACK! Ansible's playbook/play_context.py initializes the become_exe
+            # with the become method's name if it is not explicitly specified.
+            # Since in the community.general collection that name is prefixed with
+            # `community.general.`, we have to check for that default and replace
+            # it with the "short name" of the become plugin.
+            become_exe = 'doas'
```

Comment:
It doesn't work (in fact the default is already there), because Ansible's playbook/play_context.py initializes the become_exe with the become method's name if it is not explicitly specified. That happens *before* the option defaults are processed. (See also ansible/ansible#68474)

Reply:
https://github.com/ansible-collections/community.general/blob/a4322fe069aba54bc00f39018c5828d609788c47/plugins/become/doas.py#L27-L29

Reply:
Sorry, I only now realized that this only happened in the unit tests, not in "real life".